app/main.py

```diff
@@ -2,0 +1,40 @@
+class Animal:
+
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+    @classmethod
+    def __str__(cls) -> str:
+        return str([str(animal) for animal in cls.alive])
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, target: int) -> None:
```

Comment:
The `bite` method's `target` parameter is typed as `int`, which is incorrect. It should be typed as `Herbivore` since the method checks if `target` is an instance of `Herbivore`. Consider changing the type annotation to `Herbivore`.